packages/web/src/common/store/pages/audio-rewards/slice.ts

```diff
@@ -18,6 +18,39 @@ export enum ClaimStatus {
   ERROR = 'error'
 }
 
+type ClaimNone = {
+  status: ClaimStatus.NONE
+}
+
+type ClaimClaiming = {
+  status: ClaimStatus.CLAIMING
+}
+
+type ClaimWaitingForRetry = {
+  status: ClaimStatus.WAITING_FOR_RETRY
+}
+
+type ClaimAlreadyClaimed = {
+  status: ClaimStatus.ALREADY_CLAIMED
+}
+
+type ClaimSuccess = {
+  status: ClaimStatus.SUCCESS
+}
+
+type ClaimError = {
+  status: ClaimStatus.ERROR
+  aaoErrorCode: number | undefined
+}
+
+export type ClaimState =
```

Comment:
Looks great. Could save a tiny bit of typing by inlining them like so, since we never export the inermediate states anyways:
```
export type ClaimState = 
{ status: ClaimStatus.NONE } |
{ status: ClaimStatus.CLAIMING } |
...
```